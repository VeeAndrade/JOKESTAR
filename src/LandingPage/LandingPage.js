import React, { Component } from 'react';
import Joke from '../Joke/Joke';
import { getRandomJoke } from '../apiCalls';
import { setJoke } from '../Actions'
import { connect } from 'react-redux'
import './LandingPage.scss';

class LandingPage extends Component {
  constructor() {
    super();
    this.state = {
      error: null
    }
  }

  fetchRandomJoke() {
    getRandomJoke()
    .then(joke => {
      this.props.updateJoke(joke)

      this.setState({error: null})
    })
  }

  checkUser = () => {
    !this.props.username ? this.setState({
      error: '*Login to favorite this joke*'
    }) : console.log('error')
  }

  render() {
    return (
      <section className='landing-page-section'>
        <section className='joke-section'>
          <Joke joke={this.props.joke}/>
        </section>
          <section className='must-login-error-section'>
            <p className='error-msg'>{this.state.error}</p>
          </section>
        <section className='landing-action-buttons'>
          <button className = 'favorite-joke-button' onClick = {() => this.checkUser()}> Favorite </button>
          <button className='next-joke-button' onClick={() => this.fetchRandomJoke()}>Next joke</button>
        </section>
      </section>
    )
  }
}

export const mapStateToProps = state => ({
  username: state.username,
  joke: state.joke
})

export const mapDispatchToProps = dispatch => ({
  updateJoke: joke => dispatch(setJoke(joke))
})

export default connect(mapStateToProps, mapDispatchToProps)(LandingPage)