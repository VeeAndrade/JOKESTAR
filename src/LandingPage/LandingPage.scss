$faded-purple: rgba(74, 21, 160, 0.452);
$faded-blue: rgba(18, 125, 247, 0.452);
$bright-green: rgb(0, 255, 136);

* {
  z-index: 0; 
}

.landing-page-section {
  border: solid 2px $bright-green;
  background: linear-gradient($faded-purple, $faded-blue);
  height: 30rem;
  width: 60%;
  margin-top: 5rem;
  border-radius: 5px;
}

.joke-section {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 60%;
}

.landing-page-joke {
  color: #fff;
}

.landing-action-buttons {
  height: 20%;
  display: flex;
  justify-content: space-around;
}

.next-joke-button,
.favorite-joke-button {
  width: 250px;
  height: 50px;
  font-size: 1em;
  color: #fff;
  background-color: rgba(255, 255, 255, 0.205);
  border-radius: 5px;
  font-weight: bold;

  &:hover {
    cursor: pointer;
    color: rebeccapurple;
    background-color: #fff;
  }
}

.must-login-error-section {
  width: 100%;
  height: 15%;
  display: flex;
  justify-content: center;
  color: rgba(211, 11, 121, 0.986);
  font-weight: bold;
}