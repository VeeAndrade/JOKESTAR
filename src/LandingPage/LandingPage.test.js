import React from 'react';
import { LandingPage} from './LandingPage';
import { shallow } from 'enzyme';

describe('LandingPage', () => {

  let wrapper

  beforeEach(() => {
    wrapper = shallow( <LandingPage
    joke={{id: 42, setup: 'how much does a polarbear weigh?', delivery: 'enough to bresk the ice'}}
    username={'EvilCat'}
    />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
}); 