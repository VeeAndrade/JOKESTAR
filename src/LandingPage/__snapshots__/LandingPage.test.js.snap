// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should match the snapshot 1`] = `
<section
  className="landing-page-section"
>
  <section
    className="joke-section"
  >
    <Joke
      joke={
        Object {
          "delivery": "enough to bresk the ice",
          "id": 42,
          "setup": "how much does a polarbear weigh?",
        }
      }
    />
  </section>
  <section
    className="must-login-error-section"
  >
    <p
      className="error-msg"
    />
  </section>
  <section
    className="landing-action-buttons"
  >
    <button
      className="favorite-joke-button"
      onClick={[Function]}
    >
       Favorite 
    </button>
    <button
      className="next-joke-button"
      onClick={[Function]}
    >
      Next joke
    </button>
  </section>
</section>
`;
